src/cmd/mod.rs

```diff
@@ -381,7 +381,14 @@ impl<'w, 'pl> Command<'w, 'pl> {
                 ),
                 Binary::__NonExaustive => panic!("do not create __NonExaustive variants manually"),
             };
-            let mut cmd = AsyncCommand::new(crate::utils::normalize_path(&binary));
+            // This is the relative path to a command,
+            // so we need to normalize it or it won't be found if we change directories
+            // NOTE: if components.count() == 1, then this is a command that will be looked up in $PATH,
+            // so we shouldn't normalize it.
+            if binary.is_relative() && binary.components().count() > 1 {
+                binary = crate::utils::normalize_path(&binary);
+            }
```

Comment:
Hmm, this is not the logic that we should use here. `normalize_path` should always be called for binaries managed by rustwide, and never be called for global binaries.